### PR TITLE
Fix: Populate unittest inputs before rendering query, which may access the input tables

### PR DESCRIPTION
**Pull request by crericha**
- dbt models may try to hit the db during rendering, so make sure the db tables are populated before rendering the model.
- Allow unit tests to override table mappings when rendering.

### CODE REVIEW

File: sqlmesh/core/renderer.py [hunk 1]
```diff
@@ -291,7 +302,7 @@ def render(
             The rendered expression.
         """
         cache_key = _dates(start, end, latest)
-        skip_cache = bool(snapshots or expand)
+        skip_cache = bool(snapshots or table_mapping or expand)
```
Comment (izeigerman) [inline sqlmesh/core/renderer.py:305]: @tobymao I don't get why we skip cache here on snapshots, etc. We don't cache after table resolution anyway.

File: sqlmesh/core/test/definition.py [hunk 2]
```diff
@@ -152,67 +153,44 @@ def _get_rows(self, table: list[Row] | dict[str, list[Row]]) -> list[Row]:
...
 
     def runTest(self) -> None:
-        self.test_ctes()
+        # For tests we just use the model name for the table reference and we don't want to expand
+        mapping = {
+            name: _test_fixture_name(name)
+            for name in self.models.keys() | self.body.get("inputs", {}).keys()
+        }
+        query = self.model.render_query_or_raise(
+            add_incremental_filter=True,
```
Comment (izeigerman) [inline sqlmesh/core/test/definition.py:182]: This flag is no long relevant in the renderer. Let's remove it.

**Review by izeigerman — APPROVED** [review]

**Review by vchan — APPROVED** [review]

File: sqlmesh/core/renderer.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/renderer.py:305]: if these are not needed, can you remove them @crericha

Comment (crericha) [reply sqlmesh/core/renderer.py]: Confirmed not needed anymore. Removed in the latest commit to this PR.

**Pull request merged**